File: src/kwd.BoxOBlazor.Demo/Web/scripts/util/Clipboard.cs
using System.Threading.Tasks;

using Microsoft.AspNetCore.Components;
using Microsoft.JSInterop;

namespace kwd.BoxOBlazor.Demo.util
{
    public class Clipboard
    {
        private readonly IJSRuntime _proxy;
        private const string JSNamespace = "kwd.BoxOBlazor.Util.Clipboard";

        public Clipboard(IJSRuntime proxy)
        {
            _proxy = proxy;
        }

        /// <summary>
        /// Copy text content from given element to the clipboard.
        /// </summary>
        public async Task CopyFromElement(ElementReference elem)
            => await _proxy.InvokeVoidAsync(JSNamespace + ".copyFromElement", elem);

        /// <summary>
        /// Copy text to clipboard
        /// </summary>
        public async Task CopyToClipboard(string text)
            => await _proxy.InvokeVoidAsync(JSNamespace + ".copyToClipboard", text);
    }
}

File: src/kwd.BoxOBlazor.Demo/Web/scripts/util/Clipboard.ts
namespace kwd.BoxOBlazor.Util {

	export namespace Clipboard {
		//based on eg from 
		//https://www.w3schools.com/howto/howto_js_copy_clipboard.asp
		export function copyFromElement(elem: HTMLElement) {

			copyToClipboard(elem.innerText);
		}

		export function copyToClipboard(text: string) {

			let buffer = ensureClipboardBuffer();

			buffer.textContent = text;
			buffer.select();
			buffer.setSelectionRange(0, 99999);

			document.execCommand("copy");

			document.getSelection().removeAllRanges();
		}

		function ensureClipboardBuffer() {
			let buffer = document.getElementById("__internalClipboardBuffer") as HTMLTextAreaElement;
			if (!buffer) {
				buffer = document.createElement("textarea");
				buffer.id = "__internalClipboardBuffer";
				buffer.style.position = "absolute";
				buffer.style.left = "-9999px";
				document.body.append(buffer);
			}

			return buffer;
		}
	}
}